### PR TITLE
Extract request id handling to dedicated class so it can be injected without DB dependency

### PR DESCRIPTION
**Pull request by nickvergessen**
Logs before (with `query_log_file`):
```
SELECT * FROM `oc_appconfig`
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED
SET SESSION AUTOCOMMIT=1
SELECT `class`, `entity`, CAST(`events` AS CHAR) AS `events` FROM `oc_flow_operations` WHERE `events` <> :dcValue1 GROUP BY `class`, `entity`, CAST(`events` AS CHAR)
```

Logs after (with `query_log_file` and `query_log_file_requestid` === yes)
```
jjI3eJAK1Ljz1OptLtpG	SELECT * FROM `oc_appconfig`
jjI3eJAK1Ljz1OptLtpG	SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED
jjI3eJAK1Ljz1OptLtpG	SET SESSION AUTOCOMMIT=1
jjI3eJAK1Ljz1OptLtpG	SELECT `class`, `entity`, CAST(`events` AS CHAR) AS `events` FROM `oc_flow_operations` WHERE `events` <> :dcValue1 GROUP BY `class`, `entity`, CAST(`events` AS CHAR)
```

Required for more in depth debugging on parallel requests

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

File: lib/private/Server.php [hunk 1]
```diff
@@ -1040,6 +1042,13 @@ public function __construct($webRoot, \OC\Config $config) {
 		/** @deprecated 19.0.0 */
 		$this->registerDeprecatedAlias('Request', \OCP\IRequest::class);
 
+		$this->registerService(\OCP\IRequestId::class, function (ContainerInterface $c) {
```
Comment (CarlSchwan) [inline lib/private/Server.php:1045]:
```suggestion
		$this->registerService(IRequestId::class, function (ContainerInterface $c): IRequestId {
```

File: lib/private/DB/Connection.php [hunk 2]
```diff
@@ -271,11 +271,15 @@ public function executeStatement($sql, array $params = [], array $types = []): i
 	}
 
 	protected function logQueryToFile(string $sql): void {
-		$logFile = $this->systemConfig->getValue('query_log_file', '');
+		$logFile = $this->systemConfig->getValue('query_log_file');
+		$prefix = '';
+		if ($this->systemConfig->getValue('query_log_file_requestid') === 'yes') {
```
Comment (CarlSchwan) [inline lib/private/DB/Connection.php:276]: Can we put this inside the `if ($logFile !== '' && ` condition?

**Review by CarlSchwan — APPROVED** [review]
Tests needs to be adapted, but other than a few small nitpicks this looks fine to me.

In some ways, this feels a bit like duplicated work with https://github.com/nextcloud/server/pull/30823 since the requests are also stored with the requested (and arguments + execution time) there :/

We might want to discuss how we want to move forward since there is also OCP\Diagnostics too that also logs queries and having 3 DB requests loggers is maybe a bit too much

**Comment by nickvergessen** [comment]
> In some ways, this feels a bit like duplicated work with https://github.com/nextcloud/server/pull/30823 since the requests are also stored with the requested (and arguments + execution time) there :/

Happy to talk about it. The problem is the profiler only helps in an isolated testing case. There is no way to e.g. monitor the full company call with it.

> We might want to discuss how we want to move forward since there is also OCP\Diagnostics too that also logs queries and having 3 DB requests loggers is maybe a bit too much

That's also right. I need to checkout the Diagnostics stuff again what it does and can do.

**Pull request merged**